v2/AdminCore.WebApi.Tests/Controllers/HolidayControllerTests.cs

```diff
@@ -0,0 +1,368 @@
+using AdminCore.Common.Interfaces;
+using AdminCore.Constants.Enums;
+using AdminCore.DAL.Models;
+using AdminCore.DTOs.Employee;
+using AdminCore.DTOs.Event;
+using AdminCore.DTOs.EventDates;
+using AdminCore.WebApi.Controllers;
+using AdminCore.WebApi.Models.Holiday;
+using AutoFixture;
+using AutoMapper;
+using NSubstitute;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace AdminCore.WebApi.Tests.Controllers
+{
+  public class HolidayControllerTests : BaseControllerTest
+  {
+    private readonly HolidayController _controller;
+
+    private readonly IEventService _eventService;
+
+    private readonly IFixture _fixture;
+
+    private readonly IMapper _mapper;
+
+    public HolidayControllerTests()
+    {
+      _eventService = Substitute.For<IEventService>();
+      _mapper = Substitute.For<IMapper>();
+      _fixture = new Fixture();
+
+      _controller = new HolidayController(_eventService, _mapper);
+    }
+
+    [Fact]
+    public void ApproveHoliday_WhenCalled_ReturnsApprovedHoliday()
+    {
+      // Arrange
+      var approveViewModel = _fixture.Create<ApproveHolidayViewModel>();
+      var eventDto = _fixture.Create<EventDto>();
+
+      _mapper.Map<EventDto, ApproveHolidayViewModel>(Arg.Is(eventDto)).Returns(approveViewModel);
+
+      // Act
+      var result = _controller.ApproveHoliday(approveViewModel);
+
+      // Assert
+      RetrieveValueFromResult<List<HolidayViewModel>>(result);
```

Comment:
Might want to assert: approveViewModel and the return from RetrieveValueFromResult

Reply:
How do you mean?